### PR TITLE
Breaking changes for MOI v0.10

### PR DESCRIPTION
**Pull request by metab0t**
- The new `test/MathOptInterface/MOI_Wrapper.jl` tests are all passed except some conflict tests that need further investigation
- The old `test/MathOptInterface/MOI_Wrapper.jl` is renamed as `test/MOI_Wrapper_legacy.jl` to wipe out old `MOI.Test` related tests and keep handwritten tests. `test/MathOptInterface/MOI_Wrapper.jl` tests are all passed.
- Some of `test/MathOptInterface/MOI_callbacks.jl` tests failed. @joaquimg Would you like to have a look?
- I tried to solve #127 by using `XPRSaddmipsol` to set up initial value of variables but I am not sure about the correctness.

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -555,22 +555,32 @@ end
...
 
 function MOI.copy_to(dest::Optimizer, src::MOI.ModelLike; kwargs...)
-    return MOI.Utilities.automatic_copy_to(dest, src; kwargs...)
+    return MOI.Utilities.default_copy_to(dest, src; kwargs...)
 end
 
 function MOI.get(model::Optimizer, ::MOI.ListOfVariableAttributesSet)
-    return MOI.AbstractVariableAttribute[MOI.VariableName()]
+    ret = MOI.AbstractVariableAttribute[]
+    found_name = all(!isempty(info.name) for info in values(model.variable_info))
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:575]: Shouldn't this be 
```suggestion
    found_name = any(!isempty(info.name) for info in values(model.variable_info))
```

File: src/MOI/MOI_wrapper.jl [hunk 2]
```diff
@@ -555,22 +555,32 @@ end
...
+    found_start = all(info.start !== nothing for info in values(model.variable_info))
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:576]:
```suggestion
    found_start = any(info.start !== nothing for info in values(model.variable_info))
```

**Review by odow — COMMENTED** [review]
You can remove the deprecated test file entirely.

It's okay to exclude the tests for now. They can be fixed in future PRs.

File: src/MOI/MOI_wrapper.jl (hunk 1 above)
Comment (metab0t) [reply src/MOI/MOI_wrapper.jl:575]: Thanks for the catch!

**Comment by joaquimg** [comment]
Please do:

> I tried to solve Xpress ignore warm starts #127 by using XPRSaddmipsol to set up initial value of variables but I am not sure about the correctness.

in a separate PR.

**Comment by metab0t** [comment]
> 
> 
> @metab0t callbacks pass locally in master before you commits?

Yes, it passes before my commits.

**Comment by metab0t** [comment]
The callback error
Snipaste_2021-09-21_23-22-52

**Comment by joaquimg** [comment]
a few tests disappeared, can you review that?

**Comment by joaquimg** [comment]
I noticed @odow asked for the removal of the old tests.
Are those corner cases covered in the default MOI tests?

**Comment by odow** [comment]
Ah sorry. To clarify: you can remove _most_ of the deprecated test file, i.e., anything that called `MOI.DeprecatedTest`. If there were additional Xpress-specific tests, we should keep those.

**Comment by odow** [comment]
@metab0t this would be easier if I gave you access to make a PR to the `jump-dev` remote so that CI could run. 

Are you based at a university? Your GitHub profile is a little sparse...

**Comment by odow** [comment]
@metab0t, I talked to @joaquimg and I've sent you an invite to be an outside collaborator. Use the access wisely :)

The easiest thing to do is start a new PR against the jump-dev remote (CI only runs if you have access to the secret to enable Xpress to download). 

Let me know if you need help doing that.

**Comment by metab0t** [comment]
> 
> 
> Ah sorry. To clarify: you can remove _most_ of the deprecated test file, i.e., anything that called `MOI.DeprecatedTest`. If there were additional Xpress-specific tests, we should keep those.

OK, I will add those additional tests.



> 
> 
> @metab0t, I talked to @joaquimg and I've sent you an invite to be an outside collaborator. Use the access wisely :)
> 
> The easiest thing to do is start a new PR against the jump-dev remote (CI only runs if you have access to the secret to enable Xpress to download).
> 
> Let me know if you need help doing that.

Thanks! I have accepted the invitation. I will start a new PR today.

> 
> 
> @metab0t this would be easier if I gave you access to make a PR to the `jump-dev` remote so that CI could run.
> 
> Are you based at a university? Your GitHub profile is a little sparse...

Yes, I am currently a PhD student.

**Comment by metab0t** [comment]
New PR #146 is opened.
